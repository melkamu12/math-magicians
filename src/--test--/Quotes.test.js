// Quotes.test.js
import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import Quotes from '../components/Quote';

describe('Quote component', () => {
  it('should matche DOM Snapshot', async () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Quotes />
        </MemoryRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders a list of quotes', async () => {
    const data = [
      {
        quote:
          'I&apos;m ultimately a widow and a single mother, who &apos; not even getting to be a mother right now. I am so alone, it&apos;s freaky..',
        author: 'Courtney Love',
        category: 'happiness',
      },
      {
        quote: 'I was never less alone than when by myself.',
        author: 'Edward Gibbon',
        category: 'selfhelp',
      },
      {
        quote: 'Nothing makes us so lonely as our secrets.',
        author: 'Paul Tournier',
        category: 'fitness',
      },
    ];
    render(
      <div className="my-quotes">
        {data.map((item) => (
          <h3 title="quotes" key={item.author}>
            -
            {' '}
            {item.quote}
          </h3>
        ))}
      </div>,
    );
    await screen.debug();
    const quotes = await screen.findAllByTitle('quotes');

    expect(quotes).toHaveLength(3);
  });
});
