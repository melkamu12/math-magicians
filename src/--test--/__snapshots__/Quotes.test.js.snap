// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote component should matche DOM Snapshot 1`] = `
<div>
  Loading...
</div>
`;
